code/camera/camera.cpp

```diff
@@ -314,6 +314,24 @@ void camera::set_rotation_facing(vec3d *in_target, float in_rotation_time, float
 		matrix orient_buf;
 		matrix *orient = nullptr;
 
+		if (Use_model_eyepoint_for_set_camera_facing) {
```

Comment:
This is the wrong place for the change in behavior.  Since the flag is supposed to modify the functionality of `set-camera-host`, take a look at that sexp implementation and the functions called by it.

(If you still need help after that hint, I can give more specific information.)